lua/wikis/commons/Widget/Patch/Calendar.lua

```diff
@@ -97,4 +61,39 @@ function PatchCalendar:build()
 	}
 end
 
+---@return StandardPatch[]
+function PatchCalendar:_fetch()
+	local args = self.props
+	local startDate = DateExt.readTimestamp(args.sdate)
+	local year = startDate and tonumber(DateExt.formatTimestamp('Y', startDate))
+		or tonumber(args.year) or tonumber(os.date('%Y'))
+	---@cast year -nil
+
+	self.displayYear = year
+
+	return Patch.getByGameYearStartDateEndDate{
+		game = args.game,
+		startDate = startDate,
+		endDate = DateExt.readTimestamp(args.edate),
+		year = (not startDate) and year or nil,
+		limit = tonumber(args.limit),
+	}
+end
+
+---@param patches StandardPatch[]
+---@return table<integer, true>
+function PatchCalendar._getMonthsPresent(patches)
+	local monthsPresent = {}
+	Array.forEach(patches, function(patch)
+		local patchTimeStamp = patch.releaseDate.timestamp
+		-- can never be nil due to lpdb data having a valid date string
+		---@cast patchTimeStamp -nil
```

Comment:
If it never can be nil, should the anno be updated instead?

Reply:
leftover, can just kick due to using Patch.patchFromRecord